javalib/src/main/scala/java/lang/String.scala

```diff
@@ -711,6 +949,76 @@ final class _String()
     buf.toString
   }
 
+  /** Replaces special characters in this string (possibly in special contexts)
+   *  by dedicated strings.
+   *
+   *  This method encodes the general pattern of
+   *
+   *  - `toLowerCaseLithuanian()`
+   *  - `toLowerCaseTurkishAndAzeri()`
+   *  - `toUpperCaseLithuanian()`
+   *  - `toUpperCaseTurkishAndAzeri()`
+   *
+   *  @param replacementAtIndex
+   *    A function from index to `String | Null`, which should return a special
+   *    replacement string for the character at the given index, or `null` if
+   *    the character at the given index is not special.
+   */
+  @inline
+  private def replaceCharsAtIndex(replacementAtIndex: Int => String): String = {
+
+    var prep           = ""
```

Comment:
Using a `var` of type `String` is efficient in Scala.js, but not ideal in Scala Native. We should use a `java.lang.StringBuilder`. Ideally we would only allocate it lazily the first time `replacement != null`. Something like:
```suggestion
    var prep: java.lang.StringBuilder = null
```
and later
```scala
...
      if (replacement != null) {
        if (prep == null)
          prep = new java.lang.StringBuilder(len * 2)
        prep.append(this, startOfSegment, i)
        ...
```